testing/trino-testing-containers/src/main/java/io/trino/testing/containers/Minio.java

```diff
@@ -169,10 +200,74 @@ private Builder()
                     .buildOrThrow();
         }
 
+        public Minio.Builder withTLS(String privateKey, String publicCrt)
+        {
+            tls = true;
+            return withFilesToMount(Map.of(
+                    "/opt/minio/certs/private.key", privateKey,
+                    "/opt/minio/certs/public.crt", publicCrt));
+        }
+
         @Override
         public Minio build()
         {
-            return new Minio(image, hostName, exposePorts, filesToMount, envVars, network, startupRetryLimit);
+            return new Minio(image, hostName, exposePorts, filesToMount, envVars, network, startupRetryLimit, tls, truststore);
+        }
+
+        public Minio.Builder withTrustStore(String trustStorePath, String password)
+        {
+            this.truststore = Optional.of(new TrustStoreInfo(trustStorePath, password));
+            return this;
+        }
+    }
+
+    private OkHttpClient createHttpClient()
+    {
+        long fiveMinutes = TimeUnit.MINUTES.toMillis(5);
+        OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder()
+                .connectTimeout(fiveMinutes, MILLISECONDS)
+                .writeTimeout(fiveMinutes, MILLISECONDS)
+                .readTimeout(fiveMinutes, MILLISECONDS);
+
+        trustManagers.ifPresent(actualTrustManagers -> {
+            try {
+                SSLContext sslContext = SSLContext.getInstance("TLS");
+                sslContext.init(null, actualTrustManagers, new SecureRandom());
+                okHttpClientBuilder.sslSocketFactory(sslContext.getSocketFactory(), (X509TrustManager) actualTrustManagers[0]);
+            }
+            catch (KeyManagementException | NoSuchAlgorithmException e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        return okHttpClientBuilder.build();
+    }
+
+    private static TrustManager[] createTrustManagers(TrustStoreInfo trustStoreInfo)
```

Comment:
Can we reuse `S3FileSystemLoader::createTrustStore`?

Reply:
Their modules don't have common dependencies beside parent. Where do we place the common util code use by prod and test?

Reply:
initially addressed, waiting for comments